etc/spec/SPECIFICATION.md

```diff
@@ -117,12 +114,25 @@ is verified as follows:
  - The symbols . , / ? ~ ! @ # $ % ^ & * ( ) - _ = + [ ] { } \ | ' " ` ; :
 3. The username must not be reserved.
 
-The username is also validated for form, but not for reservation, as the
-username is being typed.
+A username is reserved if and only if someone is registered with that username
+or someone use it at that moment(even if that username is not registered).
+
+While the user is typing and the username is valid we have two cases.
+Either the username is registered or it is not.
+
+1. If the username is registered then a new form is appeared which it asks from the
+   user to enter the password that is related with the username. If the password
+   is correct then user is taken to the Chat screen, else an error message is
+   appeared "Ο κωδικός είναι λάθος".
+
+2. If the username is not registered then a link is appeared 'Εγγράψου με αυτό το username'
```

Comment:
I don't think it's a good idea to ask the user to register during this flow. If the user is entering a new username, then they have probably never used the application before. They want to use the application first, then register their username if they like it and once they have spent some time in it.